core/felt/felt.go

```diff
@@ -52,20 +52,19 @@ func (z *Felt) Impl() *fp.Element {
 //
 // [gnark-crypto]: https://github.com/ConsenSys/gnark-crypto/blob/master/ecc/stark-curve/fp/element.go
 func (z *Felt) UnmarshalJSON(data []byte) error {
-	s := string(data)
-	if len(s) > fp.Bits*3 {
+	if len(data) > fp.Bits*3 {
 		return errors.New("value too large (max = Element.Bits * 3)")
 	}
 
 	// we accept numbers and strings, remove leading and trailing quotes if any
-	if len(s) > 0 && s[0] == '"' {
-		s = s[1:]
+	if len(data) > 0 && data[0] == '"' {
+		data = data[1:]
 	}
-	if len(s) > 0 && s[len(s)-1] == '"' {
-		s = s[:len(s)-1]
```

Comment:
Reslicing a string causes creation of new string (unless compiler is super smart). So the optimisation is really simple: just defer string construction till the very end.